swagger.yaml

```diff
@@ -171,6 +171,27 @@ definitions:
       os:
         $ref: '#/definitions/os-information'
         description: OS lifecycle information
+      LogCollection:
+        type: object
+        additionalProperties:
+          $ref: '#/definitions/logs-collection-information'
+          description: Log collection information
+
+  logs-collection-information:
+    type: object
+    properties:
+      kind:
```

Comment:
Is `kind` needed if it is also a key in a map?

Reply:
yes, can be two logs: 

```
log_collection:
  syslog-dc-foo:
    BufferSize: 10
    kind: syslog
    syslogConfig:
      address: dc-foo:514
      proto: tcp
  syslog-dc-bar:
    BufferSize: 10
    kind: syslog
    syslogConfig:
      address: dc-bar:514
      proto: tcp
````

Reply:
OK, so the `kind` is not something user picks - it needs to come from a pre-defined set of values? In that case, WDYT about providing proper constants in `edgedevice_types.go`?

Reply:
If the structure of each log-collector is different (Based on each collector implementation), it can be represented as with
`oneOf` and each collector will have its own definition.
Then we don't need to specify `kind`